src/liboslcomp/codegen.cpp

```diff
@@ -295,7 +295,7 @@ ASTNode::coerce (Symbol *sym, const TypeSpec &type, bool acceptfloat)
     }
 
     if (sym->symtype() == SymTypeConst && sym->typespec().is_int() &&
-            type.is_floatbased()) {
+            type.is_float_based() && !type.is_array()) {
```

Comment:
Did you mean to use `is_float_based_not_array()` here?

Reply:
Nothing subtle. There were just two ways to express this and I chose based on aesthetics (and forgetfulness).

I think when I visited this file, I was toying with seeing whether I could get rid of is_floatbased entirely. But there were some others files I visted later where the complexity of the expression made it more awkward to add another clause, so I chose the "principle of least change" and just changed the one call to is_floatbased to is_float_based_notarray.

I can be more consistent if you think it's better -- either always using is_float_based_notarray whenever that's applicable (never letting it be is_float_based && !is_array), or else always doing the latter and eliminating is_float_based_notarray entirely.

Reply:
For this particular case, as with many others, I think I realized that it could never be called for the array case, so I just changed is_floatbased to is_float_based.